src/core/p5.Renderer2D.js

```diff
@@ -11,13 +11,46 @@ import './p5.Renderer';
 const styleEmpty = 'rgba(0,0,0,0)';
 // const alphaThreshold = 0.00125; // minimum visible
 
-class Renderer2D extends p5.Renderer{
-  constructor (elt, pInst, isMainCanvas) {
+class Renderer2D extends p5.Renderer {
+  constructor(elt, pInst, isMainCanvas) {
     super(elt, pInst, isMainCanvas);
     this.drawingContext = this.canvas.getContext('2d');
     this._pInst._setProperty('drawingContext', this.drawingContext);
   }
 
+  getFilterGraphicsLayer() {
+    // create hidden webgl renderer if it doesn't exist
+    if (!this.filterGraphicsLayer) {
+      // the real _pInst is buried when this is a secondary p5.Graphics
+      const pInst =
+        this._pInst instanceof p5.Graphics ?
+          this._pInst._pInst :
+          this._pInst;
+
+      // create secondary layer
+      this.filterGraphicsLayer =
+        new p5.Graphics(
+          this.width,
+          this.height,
+          constants.WEBGL,
+          pInst
+        );
+    }
+    if (
+      this.filterGraphicsLayer.width !== this.width ||
+      this.filterGraphicsLayer.height !== this.height
+    ) {
+      // Resize the graphics layer
+      this.filterGraphicsLayer.resize(this.width, this.height);
```

Comment:
I think in this case, since `filterGraphicsLayer` is a `p5.Graphics`, we can use `resizeCanvas` instead of `resize`. Unfortunately it's kinda confusing that the main canvas and graphics call it one thing but images and framebuffers call it something else :')

Reply:
oops sorry, i commited it wrong. sorry for this

Reply:
actually take a look at the test for this. I am not good at writing tests.